lib/applications/resolvers.ts

```diff
@@ -713,3 +717,131 @@ export const generateApplicationsReport: Resolver = async (_, args, { prisma })
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  const columnsSet = new Set(columns);
+
+  const applications = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: startDate,
+        lte: endDate,
+      },
+    },
+    select: {
+      rcdUserId: columnsSet.has(PermitHoldersReportColumn.UserId),
+      firstName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      middleName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      lastName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      dateOfBirth: columnsSet.has(PermitHoldersReportColumn.ApplicantDateOfBirth),
+      addressLine1: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      addressLine2: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      city: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      province: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      postalCode: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      email: columnsSet.has(PermitHoldersReportColumn.Email),
+      phone: columnsSet.has(PermitHoldersReportColumn.PhoneNumber),
+      guardianFirstName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianMiddleName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianLastName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianRelationship: columnsSet.has(PermitHoldersReportColumn.GuardianPoaRelation),
+      guardianAddressLine1: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianAddressLine2: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianPostalCode: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianCity: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianProvince: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      permit: {
+        select: {
+          rcdPermitId: true,
+          active: columnsSet.has(PermitHoldersReportColumn.UserStatus),
+        },
+      },
+      permitType: columnsSet.has(PermitHoldersReportColumn.RecentAppType),
+    },
+  });
+
+  // Formats the date fields and adds totalAmount, applicantName and rcdPermitId properties to allow for csv writing
+  const csvApplications = applications.map(application => {
+    return {
+      ...application,
+      dateOfBirth: formatDate(application.dateOfBirth),
+      applicantName:
+        application.firstName +
+        (application.middleName
+          ? ` ${application.middleName} ${application.lastName}`
+          : ` ${application.lastName}`),
+      rcdPermitId: application.permit?.rcdPermitId,
+      homeAddress: `${application.addressLine1}, 
+      ${application.addressLine2 ? `${application.addressLine2},` : ''} ${application.city}, ${
+        application.province
+      }, ${application.postalCode}`,
```

Comment:
```suggestion
      } ${application.postalCode}`,
```
Usually, we don't write a comma between province and postal code

Reply:
Addressed!